packages/snap-client/src/Client/transforms/searchResponse.ts

```diff
@@ -212,31 +219,40 @@ transformSearchResponse.result = (rawResult: rawResult): SearchResponseModelResu
 
 	if (coreFieldValues.price) coreFieldValues.price = +coreFieldValues.price;
 	if (coreFieldValues.msrp) coreFieldValues.msrp = +coreFieldValues.msrp;
-
+	if (coreFieldValues.available?.toString() === 'true') {
+		coreFieldValues.available = true;
+	} else if (coreFieldValues.available?.toString() === 'false') {
+		coreFieldValues.available = false;
+	}
 	const attributes = Object.keys(rawResult)
 		.filter((k) => CORE_FIELDS.indexOf(k) == -1)
 		// remove 'badges' from attributes - but only if it is an object
 		.filter((k) => !(k == 'badges' && Array.isArray(rawResult[k]) && typeof rawResult[k]?.[0] == 'object'))
+		.filter((k) => !(k == 'variants'))
 		.reduce((attributes, key) => {
 			return {
 				...attributes,
 				[key]: decodeProperty(rawResult[key as keyof rawResult] || ''),
 			};
 		}, {});
 
-	const children =
-		rawResult?.children?.map((child) => {
-			return {
-				attributes: {
-					...Object.keys(child).reduce((attributes, key) => {
-						return {
-							...attributes,
-							[key]: decodeProperty(child[key]),
-						};
-					}, {}),
-				},
-			};
-		}) || [];
+	if (rawResult.variants && rawResult.variants.data) {
+		rawResult.variants.data.forEach((variant) => {
+			// @ts-ignore - transforming the data
+			if (variant.core) {
+				// @ts-ignore - transforming the data
+				variant.badges = variant.core.badges;
+
+				variant.mappings = {
+					// @ts-ignore - transforming the data
+					core: variant.core,
+				};
+
+				// @ts-ignore - transforming the data
+				delete variant.core;
+			}
+		});
+	}
```

Comment:
I'm pretty sure this section is unnecessary - or could use refactoring, as there is no longer `variant.core` - instead it should come back as `variant.mappings.core`.